tools/clang_format.py

```diff
@@ -124,56 +115,101 @@ def get_clang_format_diff(filename, lines):
             )
 
 
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Execute clang-format on your working copy changes."
+    )
+    parser.add_argument(
+        "-d",
+        "--diff",
+        default="HEAD",
+        help="Git revision to diff against to get changes",
+    )
+    parser.add_argument(
+        "--accept-changes",
+        action="store_true",
+        default=False,
+        help=(
+            "If true, apply whatever changes clang-format creates. "
+            "Otherwise, just print the changes and exit"
+        ),
+    )
+    parser.add_argument(
+        "--check-all",
+        action="store_true",
+        default=False,
+        help="If true, check all whitelisted files instead of just working copy changes",
+    )
+    parser.add_argument("--verbose", "-v", action="store_true", default=False)
+    return parser.parse_args()
+
+
 def main():
     args = parse_args()
 
-    whitelisted_files = get_whitelisted_files()
+    # This flag is pervasive enough to set it globally. It makes the code
+    # cleaner compared to threading it through every single function.
+    global VERBOSE
+    VERBOSE = args.verbose
 
+    whitelisted_files = get_whitelisted_files()
     if args.check_all:
-        files_to_check = whitelisted_files
+        files_to_check = sorted(whitelisted_files)
     else:
         changed_files = get_changed_files(args.diff)
-        files_to_check = changed_files & whitelisted_files
+        files_to_check = sorted(changed_files & whitelisted_files)
 
-    if args.verbose:
+    if VERBOSE:
         print("Running clang-format on whitelisted files: ")
         for f in files_to_check:
             print(f)
 
+    # Build map with files to line ranges.
     name_to_lines = {}
-    for f in files_to_check:
-        changed_lines = get_changed_lines(f, args.diff)
-        if len(changed_lines) != 0:
-            name_to_lines[f] = changed_lines
+    if args.check_all:
+        # Include all files, with empty line range.
+        for f in files_to_check:
+            name_to_lines[f] = []
+    else:
+        # Include only changed files, with changed line ranges.
+        for f in files_to_check:
+            changed_lines = get_changed_lines(args.diff, f)
+            if len(changed_lines) != 0:
+                name_to_lines[f] = changed_lines
 
+    # If no files in the whitelist were changed, exit early.
     if len(name_to_lines) == 0:
-        return
+        print("No files detected.")
+        sys.exit()
 
+    # Build map with files to clang-format diff.
     name_to_diff = {}
     for filename, lines in name_to_lines.items():
         diff = get_clang_format_diff(filename, lines)
         if diff is not None:
             name_to_diff[filename] = diff
 
+    # If no files produced a clang-format diff, exit early.
+    if len(name_to_diff) == 0:
```

Comment:
Since this appears to occur when the code is formatted correctly, should we add:
`if VERBOSE:
      print('No formatting changes were found')
`

to distinguish this from the case where no files were detected? Might help users distinguish between errors caused by passing in files incorrectly versus not having any formatting issues.